// Most data types match their HSA counterparts, though there are some shorter ints that have been replaced
// with uint32s.

// TODO: fix grid_dim struct and fields that are arrays
syntax = "proto2";

package rhsa;

message Agent {
	required string name = 100;
	required string vendor_name = 101;
	enum Feature {
		kernel_dispatch = 1;
		agent_dispatch = 2;
	}
	required Feature feature = 102;
	enum Machine_model {
		small = 0;
		large = 1;
	}
	required Machine_model machine_model = 103;
	enum Profile {
		base = 0;
		full = 1;
	}
	required Profile profile = 104;
	enum Default_float_rounding_mode {
		zero = 1;
		near = 2;
	}
	required Default_float_rounding_mode default_float_rounding_mode = 105;
	required bool fast_f16_operation = 106;
	required uint32 wavefront_size = 107;
	required uint32 workgroup_max_dim = 108;
	required uint32 workgroup_max_size = 109;
	struct dim3 {
		uint32 x;
		uint32 y;
		uint32 z;
	}
	required dim3 grid_max_dim = 110;
	required uint32 grid_max_size = 111;
	required uint32 fbarrier_max_size = 112;
	required uint32 queues_max = 113;
	required uint32 queue_min_size = 114;
	required uint32 queue_max_size = 115;
	enum Queue_type {
		multi = 0;
		single = 1;
	}
	required Queue_type queue_type = 116;
	required uint32 node = 117;
	enum Device_type {
		cpu = 0;
		gpu = 1;
		dsp = 2;
	}
	required Device_type device = 118;
	required uint32 cache_size[4] = 119;
	required uint64 isa = 120;
	required uint32 agent_extensions[128] = 121;
	required uint32 version_major = 122;
	required uint32 version_minor = 123;
}

message QueryAgents {
	repeated Agent agents = 1;
}
